File: swift/code_wars/convert_number_to_reversed_array_of_digits.swift
func digitize(_ num: Int) -> [Int] {
    var number = num
    var digits: [Int] = []

    while number >= 0 {
        let digit = number % 10
        digits.append(digit)
        number /= 10
        if number == 0 {
            break
        }
    }

    return digits
}

File: rust/src/code_wars/convert_number_to_reversed_array_of_digits.rs
fn digitize(num: u64) -> Vec<u8> {
    let num_str = num.to_string();
    let mut digit_list = Vec::new();
    for char in num_str.chars().rev() {
        digit_list.push(char.to_digit(10).unwrap() as u8);
    }
    digit_list
}